### ISSUE TITLE
Tool for updating maps to latest version

### ISSUE THREAD

**Issue by cinaera** [teeworlds/teeworlds#1326]
Add a tool which loads old maps and saves them in a 0.7 compatible format.
Should use `CEditorMap::Load` and `CEditorMap::Save` for the update process. This allows us to remove compatiblity stuff from the map loading in the game source(e.g. https://github.com/teeworlds/teeworlds/blob/master/src/game/client/components/maplayers.cpp#L101) and is a more handy way to manually load/save maps in the editor.

**Comment by oy** [comment]
Such a tool might also be good for demos. Differences between 0.6 and 0.5 should more or less some informations. For 0.7 this probably gets difficult.

**Comment by Dune-jr** [comment]
@axblk Is your tool fully functional for 0.6 maps?

**Comment by axblk** [comment]
Yes, it can update the map format and it fixes the tile layers for the new mapres (including the new shadow mapres). Right now it is integrated in the racemod code but you can easily turn off the race support or remove the race related code.

Problem is that is does not use the `CEditorMap` class from the editor, since this one contains a lot of graphics related code which makes it unsuitable for the usage in a command line tool.
Instead of properly modifying the editor class, I'm using a stripped down version of it.

The tool also does some additional optimizations. It removes flip flags from tiles that only have one color and it removes tiles that are completely transparent.

Right now it lives here:
https://github.com/axblk/teeworlds/tree/race/src/tools

**Comment by Dune-jr** [comment]
Maybe this tool could also directly be embedded in the map editor? Something like "Import from 0.6" in the menus
